api/src/shipit_api/admin/product_details.py

```
@@ -624,7 +633,61 @@ def get_firefox_esr_next_version(releases: typing.List[shipit_api.common.models.
+    firefox_nightly_mozilla_version = FirefoxVersion.parse(shipit_api.common.config.FIREFOX_NIGHTLY)
+    current_nightly_version_major_number = firefox_nightly_mozilla_version.major_number
+    previous_nightly_version_major_number = current_nightly_version_major_number - 1
+    url_template = "https://whattrainisitnow.com/api/release/schedule/?version={version}"
+    try:
+        url = url_template.format(version=current_nightly_version_major_number)
+        async with session.get(url) as response:
+            response.raise_for_status()
+            logger.debug(f"Fetched {url}")
+            current_nightly_version_schedule = await response.json()
+        url = url_template.format(version=previous_nightly_version_major_number)
+        async with session.get(url) as response:
+            response.raise_for_status()
+            logger.debug(f"Fetched {url}")
+            previous_nightly_version_schedule = await response.json()
+    except Exception:
+        logger.info("Failed to fetch %s, %s", url)
+        raise
+    date_format = "%Y-%m-%d"
+    last_softfreeze_date = from_isoformat(previous_nightly_version_schedule["soft_code_freeze"]).strftime(date_format)
+    last_merge_date = from_isoformat(previous_nightly_version_schedule["merge_day"]).strftime(date_format)
+    releases_after_last_merge_date = sorted(
+        [
+            release
+            for release in releases
+            if release.product == Product.FIREFOX.value
+            and FirefoxVersion.parse(release.version).is_release
+            and release.status == "shipped"
+            and release.completed is not None
+            and release.completed.replace(tzinfo=datetime.timezone.utc) > from_isoformat(previous_nightly_version_schedule["merge_day"])
+        ],
+        key=lambda release: FirefoxVersion.parse(release.version),
+    )
+    first_release_after_last_merge_date = releases_after_last_merge_date[0]
```

Should we raise an error if no releases are found?

---

Yeah. It does throw an exception but its an index error. I will raise something more descriptive.